### ISSUE TITLE
False positive in unreachable_switch_case

### ISSUE THREAD

**Issue by DenchiUA** [dart-lang/linter#4496]
```dart
enum MyEnum {first, second}

typedef MyData = (
  MyEnum field1,
  MyEnum field2,
);

void main() {
  print(myFn(MyEnum.first, null));
  print(myFn(MyEnum.first, MyEnum.second));
}

MyData? myFn(MyEnum field1, MyEnum? field2) {
  return switch ((field1, field2)) {
    final MyData a => a,
    _ => null,
    // ^ unreachable_switch_case
  };
}
```

If you remove this line, dart will throw an error.
